### DISCUSSION TITLE
WasmEdge only supports one backend at the same time?

### DISCUSSION THREAD

**Discussion by KeeganJin** [WasmEdge/WasmEdge#2828] [Q&A] [answered]
Hi all, 
I noticed that the backends are exclusive based on the [instruction](https://wasmedge.org/docs/start/install/#wasi-nn-plug-in),  it said that developers can only choose and install one backend for the WASI-NN plug-in. 
**Does that also mean wasi_nn-pytorch and wasmedge_tensorflow (not using wasi_nn) plugins cannot be supported at the same time?** 
I tried `curl -sSf https://raw.githubusercontent.com/WasmEdge/WasmEdge/master/utils/install.sh | bash -s -- --plugins wasi_nn-pytorch wasmedge_tensorflow` to install two plugins, 
but then the wasi_nn-pytorch plugin in the [wasi-nn example](https://github.com/second-state/wasmedge-rustsdk-examples/tree/main/object-detection-via-wasinn) didn't work anymore.

```load plugin
[2023-09-19 17:25:21.414] [error] loading failed: invalid path, Code: 0x20
[2023-09-19 17:25:21.414] [error]     load library failed:libtorch.so: cannot open shared object file: No such file or directory
Error: Plugin(NotFound("wasi_nn"))
```

Thanks in advance. :)

**Comment by hydai** [comment, marked as answer]
If you want to have a WASI-NN plugin with multiple backends, please build from source with specifying the `-DWASMEDGE_PLUGIN_WASI_NN_BACKEND="X;Y;Z".

**Reply by KeeganJin** [reply]
Thanks a lot, I will try it!
